packages/spiderfier/services/managers/spider-manager.ts

```diff
@@ -0,0 +1,175 @@
+import {Injectable, NgZone, SkipSelf} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {OverlappingMarkerSpiderfier} from 'ts-overlapping-marker-spiderfier';
+
+import {MarkerManager} from '../../../core/services/managers/marker-manager';
+import {GoogleMapsAPIWrapper} from '../../../core/services/google-maps-api-wrapper';
+import {AgmMarker} from '../../../core/directives/marker';
+import {AgmMarkerSpider} from '../../directives/marker-spider';
+import {MarkerSpiderInstance, SpiderOptions} from '../google-spider-types';
+
+@Injectable()
+export class SpiderManager extends MarkerManager {
+  private _spiderInstance: Promise<MarkerSpiderInstance>;
+  private _resolver: Function;
+
+  constructor(protected _mapsWrapper: GoogleMapsAPIWrapper, protected _zone: NgZone, @SkipSelf() protected _markerManager: MarkerManager) {
+    super(_mapsWrapper, _zone, _markerManager);
+    this._spiderInstance = new Promise<MarkerSpiderInstance>((resolver) => {
+      this._resolver = resolver;
+    });
+  }
+
+  get instance() {
+    return this._spiderInstance;
+  }
+
+  init(options: SpiderOptions): void {
+    this._mapsWrapper.getNativeMap().then(map => {
+      const spider = new OverlappingMarkerSpiderfier(map as any, options);
+      this._resolver(spider);
+    });
+  }
+
+  addMarker(marker: AgmMarker): void {
+    this._markerManager.addMarker(marker, false);
+
+    const markerPromise = this._markerManager.getNativeMarker(marker);
+
+    this._markers.set(marker, markerPromise);
```

Comment:
This happens in the _makerManager.addMarker already

Reply:
True, but the _markerManager is a reference to the parent (notice the @SkipSelf) which means that the local _markers is not updated. By setting the _markers here too, it allows for the manager to keep a local reference to it's own markers which is used in removal/clear to only delete markers owned by the spiderfier, not by any parent (cluster/agm-map) managers)...

The update(s) to core (as explained in the intro) are to allow cascading of marker creation from "nested" marker managers up the chain (until the "core" marker manager belonging to the "agm-map" directive, that way it is possible to have a clustered-spiderfier (or any other nested (grouping) marker layer)...